tools/claude_retriever/utils.py

```diff
@@ -40,25 +41,69 @@ async def scrape_url(url: str, summarize_with_claude: bool = False,
             if anthropic_api_key is None:
                 raise ValueError("anthropic_api_key must be provided if summarize_with_claude is True")
             try:
-                content = await claude_extract(content, query, anthropic_api_key)
+                content = await claude_extract_article(content, query, anthropic_api_key)
             except Exception as e:
                 logger.warning(f"Failed to extract with Claude. Falling back to raw content. Error: {e}")
     else:
         content = missing_content_placeholder
     return content
 
 
-async def get_url_content(url: str) -> Optional[str]:
-    async with aiohttp.ClientSession() as session:
-        async with session.get(url) as response:
-            if response.status == 200:
-                html = await response.text()
-                soup = BeautifulSoup(html, 'html.parser')
-                text = soup.get_text(strip=True, separator='\n')
-                return text
+async def get_url_content(url: str, timeout: int = 10, max_size: int = 1024 * 1024) -> Optional[str]:
+    try:
+        if not is_valid_url(url):
+            logger.warning(f"Invalid URL: {url}")
+            return None
+
+        async with aiohttp.ClientSession() as session:
+            async with session.get(url, timeout=timeout, max_size=max_size, headers={'User-Agent': 'Mozilla/5.0'}) as response:
+                if response.status == 200:
+                    html = await response.text()
+                    soup = BeautifulSoup(html, 'html.parser')
+                    text = soup.get_text(strip=True, separator='\n')
+                    
+                    # Sanitize the extracted text using bleach
```

Comment:
you can never be too careful when inserting web search results into your prompts